pkg/controller/istio/istio_controller.go

```diff
@@ -0,0 +1,348 @@
+	"github.com/tigera/operator/pkg/render"
+	"github.com/tigera/operator/pkg/render/gatewayapi"
+	renderistio "github.com/tigera/operator/pkg/render/istio"
+)
+
+var (
+	// blank assignment to verify that ReconcileIstio implements reconcile.Reconciler
+	_ reconcile.Reconciler = &ReconcileIstio{}
+
+	log = logf.Log.WithName("controller_istio")
+)
+
+// Add creates a new Istio Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+//
+// Start Watches within the Add function for any resources that this controller creates or monitors. This will trigger
+// calls to Reconcile() when an instance of one of the watched resources is modified.
+func Add(mgr manager.Manager, opts options.AddOptions) error {
+	r := newReconciler(mgr, opts)
+
+	c, err := ctrlruntime.NewController("istio-controller", mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		return fmt.Errorf("failed to create istio-controller: %w", err)
+	}
+
+	// Watch for changes to primary resource Istio
+	err = c.WatchObject(&operatorv1.Istio{}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		log.V(5).Info("Failed to create Istio watch", "err", err)
+		return fmt.Errorf("istio-controller failed to watch primary resource: %v", err)
+	}
+
+	if err = utils.AddInstallationWatch(c); err != nil {
+		log.V(5).Info("Failed to create network watch", "err", err)
+		return fmt.Errorf("istio-controller failed to watch Tigera network resource: %v", err)
+	}
+
+	err = c.WatchObject(&crdv1.FelixConfiguration{}, &handler.EnqueueRequestForObject{})
+	if err != nil {
+		return fmt.Errorf("istio-controller failed to watch FelixConfiguration resource: %w", err)
+	}
+
+	return nil
+}
+
+// newReconciler returns a new reconcile.Reconciler
+func newReconciler(mgr manager.Manager, opts options.AddOptions) *ReconcileIstio {
+	r := &ReconcileIstio{
+		Client:              mgr.GetClient(),
+		scheme:              mgr.GetScheme(),
+		provider:            opts.DetectedProvider,
+		enterpriseCRDsExist: opts.EnterpriseCRDExists,
+		status:              status.New(mgr.GetClient(), "istio", opts.KubernetesVersion),
+		clusterDomain:       opts.ClusterDomain,
+		multiTenant:         opts.MultiTenant,
+	}
+
+	r.status.Run(opts.ShutdownContext)
+	return r
+}
+
+// ReconcileIstio reconciles a Istio object
+type ReconcileIstio struct {
+	client.Client
+	scheme              *runtime.Scheme
+	provider            operatorv1.Provider
+	enterpriseCRDsExist bool
+	status              status.StatusManager
+	clusterDomain       string
+	multiTenant         bool
+}
+
+// Reconcile reads that state of the cluster for a Istio object and makes changes based on the state read
+// and what is in the Istio.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
```

Comment:
```suggestion
```

I think we can just remove this boilerplate comment. It's generally understood within the operator codebase.